fuse/eval/metrics/sequence_gen/metrics_seq_gen_common.py

```diff
@@ -177,24 +194,23 @@ def _perplexity_update(
     preds = preds.detach()
     target = target.detach()
 
-    preds = preds.reshape(-1, preds.shape[-1])
-    target = target.reshape(-1)
+    preds = preds.view(-1, preds.shape[-1])
+    target = target.view(-1)
 
     if ignore_index is not None:
         mask = target.ne(ignore_index)
-        target = target.where(
-            target != ignore_index, torch.tensor(0, device=target.device)
-        )
+        target = target[mask]
```

Comment:
I'm not sure it's critical (maybe it's even better, saving memory)

Reply:
yes, I have reduced the side of pred and target, since at the end the entries corresponding to the mask are dropped. So it seems to produce the same results, but with less memory/time (not having to process the masked entries)